relinker/src/main/java/com/getkeepsafe/relinker/ApkLibraryInstaller.java

```diff
@@ -106,6 +111,37 @@ private ZipFileInZipEntry findAPKWithLibrary(final Context context,
         return null;
     }
 
+    // Loop over all APK's again in order in to detect which ABI's are actually supported.
+    // This second loop is more expensive than trying to find a specific ABI, so should
+    // only be run when no matching libraries are found. This should keep the overhead of
+    // the happy path to minimum.
+    private String[] getSupportedABIs(Context context, String mappedLibraryName) {
+        String p = "lib" + File.separatorChar + "([^\\" + File.separatorChar + "]*)" + File.separatorChar + mappedLibraryName;
+        Pattern pattern = Pattern.compile(p.toString());
+        ZipFile zipFile = null;
+        Set<String> supportedABIs = new HashSet<String>();
+        for (String sourceDir : sourceDirectories(context)) {
```

Comment:
I copied this loop from https://github.com/KeepSafe/ReLinker/blob/master/relinker/src/main/java/com/getkeepsafe/relinker/ApkLibraryInstaller.java#L67. In that location there is a retry mechanism for IO-errors. It wasn't documented why exactly this is needed, so I assumed it is to prevent a race condition at the _very_ beginning of the app startup (We have seen a few of these) and once we get to here, it should no longer be a problem, so I removed the retries. 

I'll be happy to add them here as well if you think it helps?

Reply:
From what I understand the retry logic is implemented for transient IO exceptions, we've seen this in our own application.

I agree that at this code path, the retry is no longer helpful and can be skipped.